src/Microsoft.DotNet.ProjectModel/ProjectExtensions.cs

```diff
@@ -21,7 +21,7 @@ public static string GetSourceCodeLanguage(this Project project)
         {
             foreach (var kvp in _compilerNameToLanguageId)
             {
-                if (kvp.Key == project.CompilerName)
+                if (kvp.Key == (project._defaultCompilerOptions.CompilerName))
```

Comment:
It is an internal field. And it is already marked for review. https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectModel/Project.cs#L19